newIDE/app/src/InstancesEditor/CompactInstancePropertiesEditor/CompactPropertiesSchema.js

```diff
@@ -0,0 +1,569 @@
+  onClick: (instance: gdInitialInstance) =>
+    onEditObjectByName(instance.getObjectName()),
+});
+
+const getRotationXAndRotationYFields = ({ i18n }: {| i18n: I18nType |}) => [
+  {
+    name: 'RotationX',
+    getLabel: () => i18n._(t`Rotation (X)`),
+    valueType: 'number',
+    getValue: (instance: gdInitialInstance) => instance.getRotationX(),
+    setValue: (instance: gdInitialInstance, newValue: number) =>
+      instance.setRotationX(newValue),
+    renderLeftIcon: className => <LetterX className={className} />,
+  },
+  {
+    name: 'RotationY',
+    getLabel: () => i18n._(t`Rotation (Y)`),
+    valueType: 'number',
+    getValue: (instance: gdInitialInstance) => instance.getRotationY(),
+    setValue: (instance: gdInitialInstance, newValue: number) =>
+      instance.setRotationY(newValue),
+    renderLeftIcon: className => <LetterY className={className} />,
+  },
+];
+const getRotationZField = ({
+  i18n,
+  label,
+  Icon,
+}: {|
+  i18n: I18nType,
+  label: MessageDescriptor,
+  Icon: React.ComponentType<any>,
+|}) => ({
+  name: 'Angle',
+  getLabel: () => i18n._(label),
+  valueType: 'number',
+  getValue: (instance: gdInitialInstance) => instance.getAngle(),
+  setValue: (instance: gdInitialInstance, newValue: number) =>
+    instance.setAngle(newValue),
+  renderLeftIcon: className => <Icon className={className} />,
+});
+const getXAndYFields = ({ i18n }: {| i18n: I18nType |}): Schema => [
+  {
+    name: 'X',
+    getLabel: () => i18n._(t`X`),
+    valueType: 'number',
+    getValue: (instance: gdInitialInstance) => instance.getX(),
+    setValue: (instance: gdInitialInstance, newValue: number) =>
+      instance.setX(newValue),
+    renderLeftIcon: className => <LetterX className={className} />,
+  },
+  {
+    name: 'Y',
+    getLabel: () => i18n._(t`Y`),
+    valueType: 'number',
+    getValue: (instance: gdInitialInstance) => instance.getY(),
+    setValue: (instance: gdInitialInstance, newValue: number) =>
+      instance.setY(newValue),
+    renderLeftIcon: className => <LetterY className={className} />,
+  },
+];
+const getZField = ({ i18n }: {| i18n: I18nType |}) => ({
+  name: 'Z',
+  getLabel: () => i18n._(t`Z`),
+  valueType: 'number',
+  getValue: (instance: gdInitialInstance) => instance.getZ(),
+  setValue: (instance: gdInitialInstance, newValue: number) =>
+    instance.setZ(newValue),
+  renderLeftIcon: className => <LetterZ className={className} />,
+});
+const getLayerField = ({
+  i18n,
+  layout,
+}: {|
+  i18n: I18nType,
+  layout: gdLayout,
+|}) => ({
+  name: 'Layer',
+  getLabel: () => i18n._(t`Layer`),
+  valueType: 'string',
+  getChoices: () => enumerateLayers(layout),
+  getValue: (instance: gdInitialInstance) => instance.getLayer(),
+  setValue: (instance: gdInitialInstance, newValue: string) =>
+    instance.setLayer(newValue),
+  renderLeftIcon: className => <Layers className={className} />,
+});
+const getZOrderField = ({ i18n }: {| i18n: I18nType |}) => ({
+  name: 'Z Order',
```

Comment:
at the top you have `RotationX` and here `Z Order` - maybe just a nitpick, but should we always have spaces between "words" or never?

Reply:
I'll keep human readable values in the case these are used for display (that should not be the case)